### ISSUE TITLE
nnzj in NLPModels

### ISSUE THREAD

**Issue by aldma** [JuliaSmoothOptimizers/CUTEst.jl#329]
Dear JSOptimizers,
I encountered an issue while using SlackModel, which possibly traces back to the use or definition of ``nnzj``, ``lin_nnzj`` and ``nln_nnzj`` in ``NLPModelsMeta``.

It should be reproducible with
```
using NLPModels, NLPModelsModifiers, CUTEst
nlp = CUTEstModel("HS75")
jac(nlp, nlp.meta.x0)
snlp = SlackModel(nlp)
jac(snlp, snlp.meta.x0)
```
Using julia 1.10.4, CUTEst v0.13.2, NLPModels v0.20.0, NLPModelsModifiers v0.7.2,  I have experienced the following error
```
BoundsError: attempt to access 14-element Vector{Int64} at index [11:22]
```
Looking closely at the HS75 problem and the nlp.meta structure, we have
nlp.meta.nnzj = 12
nlp.meta.lin_nnzj = 8
nlp.meta.nln_nnzj = 12
and somewhat surprisingly NLPModels.jac manages to return the Jacobian matrix. There are indeed 12 nonzeros but only 4 for linear constraints and 8 nonlinear ones.

- I would think that nnzj = lin_nnzj + nln_nnzj, but it doesn't seem to be the case. What do these properties mean?

For the SlackModel snlp we have
snlp.meta.nnzj = 14
snlp.meta.lin_nnzj = 10
snlp.meta.nln_nnzj = 12
after two slack variables are introduced (correctly). In this case, however, NLPModels.jac fails with the error shown above.

- Do you have any idea how to fix this?

Thank you!

**Comment by MaxenceGollier** [comment]
I have a similar issue on the problem `HS42`, 
when I write
```
using NLPModels, CUTEst
nlp = CUTEstModel("HS42")
```
I get the error message
```
AssertionError: nnzj == lin_nnzj + nln_nnzj
```

**Comment by amontoison** [comment]
https://github.com/ralna/CUTEst/issues/59

**Comment by amontoison** [comment]
It should be fixed with #351.
The next release will be v0.14.0.
